ksqldb-parser/src/test/java/io/confluent/ksql/parser/AstBuilderTest.java

```diff
@@ -393,7 +395,85 @@ public void shouldNotBuildLambdaFunctionNotLastArguments() {
     );
 
     // Then:
-    assertThat(e.getMessage(), containsString("mismatched input '=>' expecting {',', ')'}"));
+    assertThat(e.getMessage(), containsString("no viable alternative at input 'TRANSFORM_ARRAY(X =>'"));
+  }
+
+  @Test
+  public void shouldBuildIntervalExpression() {
+    // Given:
+    final SingleStatementContext stmt = givenQuery("SELECT TIMESTAMPADD(Col6, 5 HOURS) FROM TEST1;");
+
+    // When:
+    final Query result = (Query) builder.buildStatement(stmt);
+
+    // Then:
+    assertThat(result.getSelect(), is(new Select(ImmutableList.of(
+        new SingleColumn(
+            new FunctionCall(
+                FunctionName.of("TIMESTAMPADD"),
+                ImmutableList.of(
+                    column("COL6"),
+                    new IntervalExpression(
+                        new IntegerLiteral(5),
+                        TimeUnit.HOURS
+                    )
+                )
+            ),
+            Optional.empty())
+    ))));
+  }
+
+  @Test
+  public void shouldBuildMultipleIntervalExpression() {
+    // Given:
+    final SingleStatementContext stmt = givenQuery("SELECT TIMESTAMPADD(Col6, 5 HOURS, 6 MINUTES) FROM TEST1;");
```

Comment:
I'm not sure about using multiple intervals this way. I've seen other DBs uses one single expression for multiple intervals, like:
```
timestampadd(col6, '5 hours 6 minutes');
```

Any reason not to support it that way? I don't see other place in ksql that uses it like that.